### PR TITLE
chore(environments): add compare versions, change branch to version, …

### PR DESCRIPTION
**Pull request by irumvanselme**
…and fix documentation (by adding tag label)

Changes added

1. Compare versions.
2. Dump production database to test before running smoke tests and e2e tests.
3. Change branch to version on frontend.
4. Fix open API documentation

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -12,331 +12,39 @@ on:
...
-    with:
-      target-environment: test
 
-  # Auth
-  deploy-auth-test:
-    needs: [ setup-infrastructure-test ]
-    uses: ./.github/workflows/deploy-auth.yml
+  # Copy the target database
+  copy-target-database:
+    uses: ./.github/workflows/copy-database.yml
```
Comment (rav3n11) [inline .github/workflows/main.yml:22]: Shouldn't this only happen after the dev deployment has finished?

**Review by rav3n11 — COMMENTED** [review]
Some comments... Let me know what you think...

File: .github/workflows/main.yml (hunk 1 above)
Comment (irumvanselme) [reply .github/workflows/main.yml:22]: no, sometimes we don't deploy to dev.

Like if we want to deploy to test, we don't deploy to dev. in this case we run this simultaneously with build.

File: pipeline/scripts/compare-versions.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  if(semver.gt(_newVersion, _currentVersion)){
+    console.log(`✅ New version ${newVersion}, current version ${currentVersion}`);
+    process.exit(0);
+  } else {
+    console.log(`❌ New version ${newVersion}, current version ${currentVersion}`);
+    process.exit(1);
+  }
+}
+
+compareVersions(process.argv[2], process.argv[3])
```
[the remark this answers is not in the document]
Comment (irumvanselme) [reply pipeline/scripts/compare-versions.js:28]: yeah, sure let me

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
+name: Build
```
[the remark this answers is not in the document]
Comment (irumvanselme) [reply .github/workflows/deploy.yml:1]: yeah, thanks for spotting that out

**Pull request merged**